packages/itwinui-react/src/core/LabeledSelect/LabeledSelect.tsx

```diff
@@ -114,11 +114,7 @@ export const LabeledSelect = React.forwardRef(
     const labelId = `${useId()}-label`;
 
     return (
-      <InputGrid
-        labelPlacement={displayStyle}
-        status={status}
-        {...wrapperProps}
-      >
+      <InputGrid labelPlacement={displayStyle} {...wrapperProps}>
```

Comment:
Are you forgetting to set `data-iui-status` here? Also same in `InputGroup`

Reply:
I don't think it is needed in `LabeledSelect` and `InputGroup`, since the outline for `LabeledSelect` and general functioning of status for `InputGroup` work correctly before and after this PR.

Is it better to still add `data-iui-status` to these two components for consistency/correctness?

Reply:
Then I guess I'm confused why in `LabeledInput` you need to set **both** `<InputGrid data-iui-status={status}>` and `<Input status={status}>`. Is the latter needed?

Reply:
seems like there's some duplication here

https://github.com/iTwin/iTwinUI/blob/5d8986451e10101085c98a084fcc4a67b0d76d1e/packages/itwinui-css/src/input-container/input-container.scss#L34-L84

https://github.com/iTwin/iTwinUI/blob/5d8986451e10101085c98a084fcc4a67b0d76d1e/packages/itwinui-css/src/input-container/input-container.scss#L272-L302

Reply:
Yes, looks like passing `status` to `Input` was not required since the focus outline and the bottom status line still appear correctly on the input. Also, all `react-workshop` test images are still passing. So, I stopped passing `status` to the `Input`. (https://github.com/iTwin/iTwinUI/pull/1753/commits/35fe942b2c1fa42c8f51f4da62b5825e0ae7e78c)

Reply:
Three suggestions then:
1. Lets pass `data-iui-status` to all instances of `iui-input-grid` just to be safe
2. Lets try to remove the `status` passed to `Select` (inside `LabeledSelect`) and see what happens
3. Lets try to fix the CSS duplication by using a common selector. e.g.:
   ```scss
   @each $status in informational, positive, warning, negative {
     :is(.iui-input-grid, .iui-input, .iui-input-with-icon > .iui-select-button) {
       &:where([data-iui-status='#{$status}']) {
   ```

Reply:
also, unit tests need to be fixed

Reply:
> Lets try to fix the CSS duplication by using a common selector

I tried to remove the duplication by creating a new mixin, `iui-input-status`. (https://github.com/iTwin/iTwinUI/pull/1753/commits/12c8b67f9f2ac964747bfb816ddf7f602875fe00).

When testing, I realized that there were 3 cases of applying status styling to an input:
1. `data-iui-status` on `.iui-input`
2. `data-iui-status` on `.iui-input-with-icon > .iui-select-button`
3. `data-iui-status` on `.iui-input-grid .iui-input`

I modified a few existing css-workshop demos in `input.html` and `select.html` to test cases 2 & 3 (https://github.com/iTwin/iTwinUI/pull/1753/commits/1b9e70a1034e9932fc5b4821f0d387ae0d503784). This also replaces some unnecessary `iui-{status}` classes with `data-iui-status={status}` data attributes. As a result, the missing bottom status colored border on Select with status came back.

To test case 1, I created three new simple `.iui-input` without a `.iui-input-grid` parent (https://github.com/iTwin/iTwinUI/pull/1753/commits/18a67da9950ffe2692d6ad093f53722588bf87d0). I had to rearrange the groups of inputs a little to prevent the two sets of status inputs coming one after the other and thus creating a confusing demo. I am open to ideas to better test it without convoluting the demos too much.

Reply:
Creating a mixin does not remove duplication in the final CSS if you are still using the mixin in two different places. This is why I suggested using a common selector in my [comment above](https://github.com/iTwin/iTwinUI/pull/1753#discussion_r1450724284).

Reply:
> Lets pass `data-iui-status` to all instances of `iui-input-grid` just to be safe

Sounds good, added `data-iui-status` to all `InputGrid`/`iui-input-grid` + `status` (i.e. excluding DatePickerInput). (https://github.com/iTwin/iTwinUI/pull/1753/commits/1003d5b98f8fd6bf37f8d3fee735b34100da43f2)

> Lets try to remove the `status` passed to `Select` (inside `LabeledSelect`) and see what happens

After removing `status` passed to `Select` inside `LabeledSelect`, the focus outlines worked correctly and the failed images didn't have any breaking change. So, I proceeded with removing `style={style}` (https://github.com/iTwin/iTwinUI/pull/1753/commits/dbd9c75bd209d7edb66541d35aa6e0938f2b7db2).

I also fixed the resulting failing unit tests (https://github.com/iTwin/iTwinUI/pull/1753/commits/acc68ad5f1d7c0b693636dcf558a6ef4d5e594ed)

Reply:
> Creating a mixin does not remove duplication in the final CSS if you are still using the mixin in two different places. This is why I suggested using a common selector in my [comment above](https://github.com/iTwin/iTwinUI/pull/1753#discussion_r1450724284).

When I first tried to use the common selector, I realized that the cases were a bit involved to combine. Because in the `iui-input-grid` case, the selector was `&[data-iui-status={status}] .iui-input`:

<img width="473" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/4878f51f-a973-484e-bbe0-812e7bcc550c">

… whereas in the `.iui-input` & `.iui-input-with-icon > .iui-select-button` cases, the selector was`&[data-iui-status={status}]`:

<img width="504" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/8e273073-76c5-4d79-a00c-da7158ba2e08">
<img width="434" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/36e0d5fb-e097-4a97-8388-d686a77185a8">

Since `data-iui-status` was applied on different elements, it was a bit involved to combine these selectors and use the same `@each` loop.

Reply:
Does something like this work? Basically matching the selector against both itself as well as against descendants. 

```scss
:is(.iui-input-grid, .iui-input, .iui-input-with-icon > .iui-select-button) {
  @each $status in informational, positive, warning, negative {
    &:where([data-iui-status='#{$status}']) {
      :where(.iui-input, .iui-select-button),
      &:where(.iui-input, .iui-select-button) {
        // …
      }

      :where(.iui-input),
      &:where(.iui-input) {
        // …
      }

      :where(.iui-select-button),
      &:where(.iui-select-button) {
        // …
      }
    }
  }
}
```

Reply:
Yes, it worked, but surprisingly, it actually produced a larger `input-container.css` file that the existing mixins method.

I guess that is because the `iui-input-status` mixin has multiple selector branches inside it. Thus, `&` is written multiple times for each iteration of the loop. Since the `&` is more complex in the `:is(…)` than in the simpler existing selectors, each generated css selector is much longer when using the combined `:is(…)` selector. 

Thus, even though the combined scss selector may produce fewer css selectors, each selector is so long that the generated css is actually larger than when having more but smaller selectors.

I played around with a few similar combined `:is(…)` selectors and pasted the generated `input-container.css` file in this [Stackblitz playground](https://stackblitz.com/edit/github-cjgv2b?file=src%2Fdemo.scss).

Reply:
Maybe i didn't explain my suggestion fully, but essentially you're using it wrong by writing all the code inside a mixin and still using the mixin multiple times.

The purpose of those different selectors is to allow branching for small parts of the code.

I'm pasting the full code below if you want to try it out. It should be smaller than the original.

<details>
<summary>input-container.scss</summary>

```scss
// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
// See LICENSE.md in the project root for license terms and full copyright notice.
@use '../mixins' as *;

.iui-input-grid {
  display: grid;
  row-gap: var(--iui-size-2xs);
  grid-template-columns: 1fr;
  grid-template-rows: auto;
  align-items: center;

  &[data-iui-label-placement='inline'] {
    grid-template-rows: 1fr auto;
    grid-template-columns: auto 1fr;

    .iui-input-label {
      margin: 0 var(--iui-size-m) 0 0;

      &.iui-required {
        margin-inline-end: calc(var(--iui-size-2xs) * 1.5 - 1px);
      }
    }

    .iui-input-group {
      flex-direction: row;
      gap: var(--iui-size-m);
    }
  }

  .iui-status-message {
    grid-column-end: -1;
  }
}

:is(.iui-input-grid, .iui-input, .iui-input-with-icon > .iui-select-button) {
  @each $status in positive, warning, negative {
    &:where([data-iui-status='#{$status}']) {
      :where(.iui-input, .iui-select-button),
      &:where(.iui-input, .iui-select-button) {
        padding-block-end: calc(var(--iui-size-2xs) - 1px);

        &[data-iui-size='small'] {
          padding-block-end: 0;
        }

        &[data-iui-size='large'] {
          padding-block-end: calc(var(--iui-size-xs) - 1px);
        }
        // #endregion

        border-block-end: var(--iui-size-3xs) solid var(--iui-color-border-#{$status});

        &:focus {
          border-block-end: var(--iui-size-3xs) solid var(--iui-color-border-#{$status});
          outline: 2px solid var(--iui-color-text-#{$status});
          outline-offset: -2px;
        }

        &:hover {
          --_iui-input-border-color: var(--iui-color-border-#{$status});
          --_iui-input-with-icon-hover-border-color: var(--iui-color-border-#{$status});
        }
      }

      :where(.iui-select-button),
      &:where(.iui-select-button) {
        border-block-end: transparent;

        &::after {
          content: '';
          inline-size: 100%;
          position: absolute;
          inset-block-end: 0;
          inset-inline-start: 0;
          block-size: var(--iui-size-3xs);
          background-color: var(--iui-color-border-#{$status});
        }
      }
    }
  }
}

.iui-input-label {
  font-weight: var(--iui-font-weight-semibold);
  color: var(--_iui-input-label-color);

  &.iui-required {
    &::after {
      content: '*';
      margin-inline-start: var(--iui-size-2xs);
      color: var(--iui-color-text-negative);
    }
  }

  &[data-iui-disabled='true'] {
    --_iui-input-label-color: var(--iui-color-text-disabled);
  }

  cursor: default;

  &:is(label) {
    cursor: pointer;

    &[data-iui-disabled='true'] {
      cursor: not-allowed;
    }
  }
  display: block;

  &.iui-inline {
    margin: 0 var(--iui-size-m) 0 0;

    &.iui-required {
      margin-inline-end: calc(var(--iui-size-2xs) * 1.5 - 1px);
    }

    display: inline-flex;
    align-items: center;
  }
}

// ----------------------------------------------------------------------------

.iui-input-flex-container {
  --_iui-input-background-color: var(--iui-color-background);
  --_iui-input-border-color: var(--iui-color-border-foreground);
  --_iui-input-accent-border-color: var(--iui-color-border-accent);
  --_iui-input-text-color: var(--iui-color-text);
  --_iui-input-padding-block: var(--iui-size-2xs);
  --_iui-input-min-height: var(--iui-component-height);

  display: inline-flex;
  align-items: center;
  background-color: var(--_iui-input-background-color);
  border-radius: var(--iui-border-radius-1);
  color: var(--_iui-input-text-color);
  position: relative;
  block-size: var(--_iui-input-min-height);
  inline-size: 100%;
  isolation: isolate;
  min-block-size: var(--_iui-input-min-height);
  block-size: fit-content;
  line-height: var(--iui-size-l);

  // Adding before pseudo-element to make inner elements more centered
  &::before {
    content: '';
    position: absolute;
    inset: 0;
    border: 1px solid var(--_iui-input-border-color);
    border-radius: var(--iui-border-radius-1);
    transition: border-color var(--iui-duration-1) ease-out;
  }

  &:hover {
    --_iui-input-border-color: var(--iui-color-border-foreground-hover);
  }

  &:has(input:focus-visible),
  &:has(textarea:focus-visible) {
    outline: 2px solid var(--_iui-input-accent-border-color);
    outline-offset: -2px;

    // Added to make sure border color is not visible with outline
    &::before {
      border-color: transparent;
    }
  }

  @supports not selector(:has(+ *)) {
    &:where(:focus-within) {
      outline: 2px solid var(--_iui-input-accent-border-color);
      outline-offset: -2px;

      // Added to make sure border color is not visible with outline
      &::before {
        border-color: transparent;
      }
    }
  }

  input,
  textarea {
    box-sizing: border-box;
    background: transparent;
    border: none;
    font: inherit;
    z-index: 2;
    min-inline-size: var(--_iui-input-min-height);
    min-block-size: var(--_iui-input-min-height);
    flex: 999;
    padding-inline: var(--iui-size-s);
    padding-block: var(--_iui-input-padding-block);

    &:where(:not(:first-child)) {
      padding-inline-start: var(--iui-size-2xs);
    }

    &::placeholder {
      color: var(--iui-color-text-placeholder);
      font-weight: var(--iui-font-weight-light);
      opacity: 1;
    }

    &:focus {
      outline: none;
    }
  }

  &[data-iui-disabled='true'] {
    --_iui-input-background-color: var(--iui-color-background-disabled);
    --_iui-input-border-color: var(--iui-color-border-disabled);
    --_iui-input-with-icon-hover-border-color: var(--iui-color-border-disabled);
    cursor: not-allowed;
  }

  &[data-iui-size='small'] {
    --_iui-input-padding-block: 0;
    --_iui-input-min-height: var(--iui-component-height-small);
  }

  &[data-iui-size='large'] {
    --_iui-input-padding-block: var(--iui-size-xs);
    --_iui-input-min-height: var(--iui-component-height-large);
    font-size: var(--iui-font-size-2);
  }

  @each $status in positive, warning, negative {
    &[data-iui-status='#{$status}'] {
      --_iui-input-accent-border-color: var(--iui-color-border-#{$status});

      &::before {
        border-block-end-width: var(--iui-size-3xs);
        border-block-end-color: var(--_iui-input-accent-border-color);
      }

      &:hover {
        --_iui-input-border-color: var(--iui-color-border-#{$status});
      }
    }
  }
}

.iui-input-group-wrapper {
  &[data-iui-label-placement='inline'] {
    display: flex;
    flex-direction: row;

    .iui-status-message {
      margin-inline-start: var(--iui-size-m);
    }
  }
}

.iui-input-group {
  display: flex;
  flex-direction: column;

  .iui-toggle-switch-wrapper {
    padding: calc(var(--iui-size-s) * 0.5) 0;
  }

  .iui-radio-wrapper,
  .iui-checkbox-wrapper {
    min-block-size: var(--iui-size-l);
  }
}

```
</details>

Reply:
Ohh, I see, I implemented your suggestion (https://github.com/iTwin/iTwinUI/pull/1753/commits/13d3e14daf8c40155bc6059f547c1f0ce48ef35e). I also added the text selection styles (https://github.com/iTwin/iTwinUI/pull/1753/commits/ab812b16bd7c80eb946f1cadd7e66c82e998d067) and some previous region comments (https://github.com/iTwin/iTwinUI/pull/1753/commits/002ea81e4a83c7045404ce9fc9a5499b82817abd)

I noticed that it did reduce the amount of produced css. But one reason for that could also be that your suggestion is not supporting `informational` that was previously supported. Should we continue supporting `informational`?

https://github.com/iTwin/iTwinUI/blob/8c5da6e200ecdcebca499baf28c7afa5660c5408/packages/itwinui-css/src/input-container/input-container.scss#L273-L275

Reply:
I'm ok with removing `informational` because we don't support it in React anyway

Reply:
**update**: Slightly refactored the selectors in [`3deac83` (#1753)](https://github.com/iTwin/iTwinUI/pull/1753/commits/3deac837b3a5da4b032da5b446ab3d2ddb09e5e4) and left out `informational`

Reply:
> I'm ok with removing `informational` because we don't support it in React anyway.
> 
> edit: It's technically a breaking change in CSS, but it's very minor and I see it more like correcting a "mistake".

Sounds good, added a changeset for leaving out `informational` (https://github.com/iTwin/iTwinUI/pull/1753/commits/3e55bb85bd6d8db658946d46c47d3761fb371696)